add a connected event

maggu2810 opened the pull request:
The listener of the chromecast is not notified if the connection has been closed (e.g. if the read thread identify that the connection has been closed).
This will add an event if the "closed" member change the state.

maggu2810 commented:
I moved the "closed = false" in front of the read thread start, because the reader thread could call "close();".
At very strange timing this could lead to a "close()" of the reader thread and a "closed = false" because the method is still not executed completely.

vitalidze commented:
I think it's incorrect to put `CLOSED` event as the spontaneous event type. The spontaneous events are sent from the chromecast device when you don't expect them. When connection is closed it should be a separate type of event. I don't see any point of storing the `Boolean` flag here since it looks like it will always be `FALSE`.

But I agree with your point about moving `closed = false` before the reader thread starts.

P.S. why have you closed the pull request?

maggu2810 commented:
I have to change the implementation a little bit.

First one question: As you state that I should keep conformity, could you please apply the same formatting rules for all your files. It is hard for me to change the code if there is different format used in different files. The most files I have inspected now are using spaces for indentation, but there are also files (at least one) that is using spaces.

At the moment I implement a Chromecast binding for the Eclipse SmartHome framework, so it could be used in all products that are using this framework (e.g. openHAB 2).
Do you prefer we discuss all the changes to get them upstream and implemented in a way that fulfills our both needs? Or do you not want to merge such stuff (e.g. breaking the current public interface) just for one implementation usage?

I cannot move the the notification in front of the thread start. If I call e.g. getStatus() in the online notification (in the listener) the reader thread must already be started.
So I am using now synchronized blocks to ensure the connect() and close() function are not handled concurrent.
The current implementation of ChromeCast's connect function creates a channel if necessary. The channel calls connect() in its constructor. So I listener for the "connected" is called before the newly created channel is assigned to the channel member. If you call getStatus() now, a new channel is created (because the current one is still null), ...
I changes the Channel handling, so it is created once ("final" in ChromeCast now) and split the connection from that constructor.

My current master contains all the changes I implemented yesterday:
https://github.com/vitalidze/chromecast-java-api-v2/compare/master...maggu2810:master

If you want to integrate, could you have a look at this please?

If you would like to keep your source tree untouched, I can implement all that stuff in my private fork without further PRs.

If you are interested how I use the current state (my master) of your library, you could have a look at https://github.com/maggu2810/shk/blob/chromecast/bundles/shk-binding-chromecast/src/main/java/de/maggu2810/shk/bundles/chromecast/internal/ThingHandlerChromecast.java

maggu2810 commented:
As I got no reply and need a version with some changes, I work on my fork now.
If you are interested in the changes you could have a look at:
https://github.com/maggu2810/shk/compare/a8fc8dec8deae0f579789637fa3ee4e040974549...944a13dd60633e63e55a7bc99256129a7976d58c
